### PR TITLE
Allow for a custom form for manual patient creation

### PR DESCRIPTION
**Pull request by paulfalgout**
Shortcut Story ID: [sc-43818]

```json
{
  "id": "patient_creation_form",
  "attributes": {
    "value": {
      "form_id": "11111",
      "submit_text": "Continue to Form 11111"
    }
  }
}
```

### CODE REVIEW

File: test/integration/services/patient-quick-search.js [hunk 1]
```diff
@@ -226,4 +227,40 @@ context('Patient Quick Search', function() {
...
+    cy
+      .get('.modal')
+      .as('searchModal')
+      .find('.patient-search__input')
+      .type('None')
+      .wait('@routeEmptyPatientSearch');
+
+    cy
+      .get('@searchModal')
+      .should('contain', 'No results match your query')
```
Comment (nmajor25) [inline test/integration/services/patient-quick-search.js:258]: Doesn't really matter that much, but could be `No results match your query.` (period included at the end). To match the `i18n` file.

Comment (nmajor25) [inline test/integration/services/patient-quick-search.js]: Doesn't really matter that much, but could be `No results match your query.` (period included at the end). To match the `i18n` file.

File: src/js/apps/globals/search_app.js
Comment (nmajor25) [inline src/js/apps/globals/search_app.js:1]: Not caused by this PR, but something I've noticed when testing.

The `No results match your query.` is often shown before the api request is finished. Would we prefer to show the `Searching...` loading text whenever there's a pending api request?

https://github.com/RoundingWell/care-ops-frontend/assets/35355575/de82a1b4-e0ae-4961-be14-a06d9f37c5c3

File: src/js/apps/globals/nav_app.js [hunk 2]
```diff
@@ -398,8 +410,14 @@ export default RouterApp.extend({
         patientModal.disableSubmit();
         patient.saveAll(patientClone.attributes)
           .then(({ data }) => {
-            Radio.trigger('event-router', 'patient:dashboard', data.id);
             patientModal.destroy();
+
+            if (patientFormId) {
+              Radio.trigger('event-router', 'form:patient', data.id, patientFormId);
```
Comment (nmajor25) [inline src/js/apps/globals/nav_app.js:416]: When we redirect to the patient form, this is what's shown for the breadcrumb link. A link to the patient dashboard. I think that makes sense...?

Screenshot 2023-12-11 at 4 09 59 PM

**Review by nmajor25 — APPROVED** [review]
Looks good 👍 . Everything worked well when testing it locally.

I just added a few comments/thinking out loud things.

File: src/js/apps/globals/nav_app.js (hunk 2 above)
Comment (paulfalgout) [reply src/js/apps/globals/nav_app.js:416]: this is the desired behavior

**Comment by paulfalgout** [comment]
@nmajor25 I pulled out the fix for the no results issue.  I'm going to have to follow up with that big

**Comment by nmajor25** [comment]
@paulfalgout sounds good 👍 that technically was an already existing bug/issue.

**Pull request merged**